test/e2e/subscription_e2e_test.go

```diff
@@ -2743,7 +2746,7 @@ func checkDeploymentHasPodConfigNodeSelector(t GinkgoTInterface, client operator
 	return nil
 }
 
-func checkDeploymentWithPodConfiguration(t GinkgoTInterface, client operatorclient.ClientInterface, csv *operatorsv1alpha1.ClusterServiceVersion, envVar []corev1.EnvVar, volumes []corev1.Volume, volumeMounts []corev1.VolumeMount, tolerations []corev1.Toleration, resources *corev1.ResourceRequirements) {
+func checkDeploymentWithPodConfiguration(t GinkgoTInterface, client operatorclient.ClientInterface, csv *operatorsv1alpha1.ClusterServiceVersion, envVar []corev1.EnvVar, volumes []corev1.Volume, volumeMounts []corev1.VolumeMount, tolerations []corev1.Toleration, resources *corev1.ResourceRequirements) error {
```

Comment:
Quick note: it looks like we might be able to lightly refactor this helper function and remove the GinkgoTInterface parameter by removing any mentions of the require package throughout the function's flow, and given we already changed a couple of the require statements to be error checks throughout this PR, it shouldn't be all that painful. It would be nice to migrate any testify statements to gomega expressions throughout the e2e suite longer term.

I think making the required changes would be pretty quick, but it also increases the scope of this PR, so I'll leave that up to whether you want to tackle that here.

Reply:
I'll look into it.  Thanks!